pkg/operator/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -86,6 +91,16 @@ func newReconciler(mgr manager.Manager, actuator actuator.Actuator, platType con
 	}
 }
 
+// isFutureCredRequestSecret is used to identify if a given secret namespace + name is one we're expecting in a future
+func isFutureCredRequestSecret(namespace, name string) bool {
+	for _, nsn := range awsactuator.GetUpcomingCredSecrets() {
```

Comment:
Should we just add GetUpcomingSecrets to the actuator interface and make this less AWS-specific?

Reply:
Done.